testutil/checkers.go

```diff
@@ -23,13 +23,46 @@ import (
 	"bytes"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"reflect"
 	"regexp"
 	"strings"
 
 	"gopkg.in/check.v1"
 )
 
+type filePresenceChecker struct {
+	*check.CheckerInfo
+	present bool
+}
+
+// FilePresent verifies that the given file exists.
+var FilePresent check.Checker = &filePresenceChecker{
+	CheckerInfo: &check.CheckerInfo{Name: "FilePresent", Params: []string{"filename"}},
+	present:     true,
+}
+
+// FileAbsent verifies that the given file does not exist.
+var FileAbsent check.Checker = &filePresenceChecker{
+	CheckerInfo: &check.CheckerInfo{Name: "FileAbsent", Params: []string{"filename"}},
+	present:     false,
+}
```

Comment:
One more idea: add optional octal mode attribute, and FileMode checker?

Reply:
+1